pyccel/parser/semantic.py

```diff
@@ -812,7 +812,9 @@ def _handle_function(self, expr, func, args, **settings):
         =======
         new_expr : FunctionCall or PyccelInternalFunction
         """
-
+        if isinstance(func, FunctionDef):
+            if 'kernel' in func.decorators:
```

Comment:
why do you need 2 lines for this? Why not:
```suggestion
        if isinstance(func, FunctionDef) and 'kernel' in func.decorators:
```

Reply:
Is there a way to add the kernel launch configuration? Why is the error always raised?

Reply:
Did you mean that we should define default values for the parameters blocks per grid and threads per blocks if they are not specified, instead of raising an error? Could you please provide more clarification on this idea?

Reply:
> Did you mean that we should define default values for the parameters blocks per grid and threads per blocks if they are not specified, instead of raising an error?

As I understand it, currently an error is raised whether parameters are specified or not. It seems strange to me to add code to handle a case that is not yet implemented

Reply:
Ah ok! I understand. Sorry I thought I was looking for arguments to the `@kernel` decorator. So the blocks per grid and threads per block are provided by the index in the function call?

Reply:
They should be provided as shown in the example below:
`launch_kernel[blocks_per_grid, threads_per_block](arguments...)`